### DISCUSSION TITLE
S2Earth::ToAngle(distance) Whether the parameter distance has a range limit？

### DISCUSSION THREAD

**Discussion by LemonLiTree** [google/s2geometry#307] [Q&A]
https://github.com/google/s2geometry/blob/master/src/s2/s2earth.h

62 static constexpr S1Angle ToAngle(const util::units::Meters& distance);

Whether the parameter distance has a range limit？

Another problem：
lng = 0 , lat = 0
through to_s2point(lng, lat, &center); get center

radius_meter = 1
through S1Angle radius = S2Earth::ToAngle(util::units::Meters(radius_meter)); get radius

Calculate the area of S2Cap：
new S2Cap(center, radius).GetArea()

the result is 7.7398634578907188e-14 ,Is this correct?

**Comment by jmr** [comment]
> Whether the parameter distance has a range limit？

No limit, it just converts meters to radians.

> the result is 7.7398634578907188e-14 ,Is this correct?

I have no reason to believe it's wrong. Do you?

**Reply by LemonLiTree** [reply]
> > Whether the parameter distance has a range limit？
> 
> No limit, it just converts meters to radians.
> 
> > the result is 7.7398634578907188e-14 ,Is this correct?
> 
> I have no reason to believe it's wrong. Do you?

What is the unit of area, I just think the result is very small (I think the unit of Meters is meters)

**Comment by jmr** [comment]
It's [steradians](https://en.wikipedia.org/wiki/Steradian). I'll add that to the docs.

**Reply by LemonLiTree** [reply]
I see
